src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppresswithnearbytextfilter/InputSuppressWithNearbyTextFilterMessagePattern.xml

```diff
@@ -20,7 +20,7 @@ maximum = (default)0
 ignoreComments = (default)false
 
 
-*/
+-->
 // violation 9 lines above 'Line matches the illegal pattern .*'
 
 <?xml version="1.0"?>
```

Comment:
This is not valid xml .
`<?xml` must be first line.
`// violation` not allowed content.

I loosing reason of update. What is a benefit ?

Reply:
> This is not valid xml .
> ?xml must be first line.

Yeah, but the rule of having the config on top will be broken if we go this path. Depends on which you believe is more-important - having config always on top or 100% validity.

> I loosing reason of update. What is a benefit ?

To make test file valid with native file extension comments. Moreover, helps when editing in an IDE.

> `// violation` not allowed content

Agreed. It is a separate issue at https://github.com/checkstyle/checkstyle/issues/13233. We can deal with that later.

Reply:
We just treat all not java files as text file.
Extension doesn't matter at all.

We can rename all such SQL, bash, xml file to be `some-name.xml.txt` and keep it clear it is txt with xml flavor/content, but it is txt.
It is txt, as we do not parse content as we do in java files. All of them are plain text.

And in plain text files we do have some lines that similar to java comments. But many other languages use such comments too. So `/*` is our start marker for special content.

One more time: I probably asked a long time ago to use xml and SQL and bash files just to show user that we can do something in text files. All of them are plain text files.

Reply:
> This is not valid xml . `<?xml` must be first line.

I do not think this is accurate for xml v1. 

As for comments, we could add one more forward slash to make a valid comment. However, I do not want to waste a bunch of time debating this or tweaking stuff. If we cannot reach agreement quickly and implementation easily, we should move ahead with imaginary input file extensions.

Reply:
> with imaginary input file extensions.

not imaginary, they are `.txt`. and we can name it  as we want:
`InputSuppressWithNearbyTextFilterMessagePattern-xml.txt`
`InputSuppressWithNearbyTextFilterMessagePatternXml.txt`
`InputSuppressWithNearbyTextFilterMessagePattern.xml.txt`

Reply:
Better to do in separate issue https://github.com/checkstyle/checkstyle/issues/13267